src/login/adal/DeviceTokenCredentials2.ts

```diff
@@ -0,0 +1,33 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AccessToken, TokenCredential } from "@azure/core-auth";
+import { Constants as MSRestConstants, WebResource } from "@azure/ms-rest-js";
+import { DeviceTokenCredentials } from '@azure/ms-rest-nodeauth';
+import { TokenResponse } from "adal-node";
+
+/**
+ * Token that is forward compatible with track 2 Azure SDK for Node.js
+ * `getToken: Promise<AccessToken>` is required for use with T2 Azure SDK, but doesn't
+ * affect T1 SDKs as those require `signRequest`
+ * `DeviceTokenCredentials` forces `getToken` to return `TokenResponse` so this is to
+ * overwrite that implementation
+ */
+export class DeviceTokenCredentials2 extends DeviceTokenCredentials implements TokenCredential {
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	public async getToken(): Promise<any> {
+		const tokenResponse = <AccessToken & TokenResponse>(await this.getTokenFromCache(this.username));
```

Comment:
I think it's worth discussing this offline. I thought I understood the concerns brought up earlier in the thread but now I fear I may not 🧐

* Won't `tokenResponse` always just be type `TokenResponse` here? The type of the object after `Object.assign` is `AccessToken & TokenResponse` since we manually add the properties required for `AccessToken`. I know you brought up an issue with the return type here but I'm confused why it doesn't work with a `AccessToken & TokenResponse` return type?
* Is there a reason you're not using `super.getToken()` here?

Reply:
I must be going bonkers here.  Before, when I tried to have the return type be `AccessToken` it was throwing an error.  I guess I never tried changing the return type when I started returning `AccessToken & TokenResponse`.

And yeah, it'll always be `tokenResponse`.  Sorry for the confusion, I originally assigning the properties directly so I had to specify it was an `AccessToken` before I could set the properties, but then I changed it to an `Object.assign`

And the reason I'm not using `super.getToken` is because it didn't occur to me 😅